### PR TITLE
chore(core): split out validators 🙀

### LINKED ISSUE #7136: feat(core): ldml: improve validator 🙀
- validate all section ranges
- validate metadata?
- more sample files, perhaps fuzzing

Furthering #7124 

#5015

### PR DESCRIPTION
**Pull request by srl295**
Fixes #7136

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by srl295** [comment]
rebased

**Comment by mcdurdin** [comment]
Note, perhaps use `Fixes #7136` or `Closes #7136` in your description so we get a 'fixing' relationship?

File: core/src/kmx/kmx_plus.cpp [hunk 1]
```diff
@@ -46,33 +50,102 @@ validate_section_name(KMX_DWORD ident) {
...
-validate_kmxplus_keys(const uint8_t* /*data*/, const COMP_KMXPLUS_KEYS* keys) {
-  if(keys == nullptr) {
-    KMXPLUS_PRINTF(("! could not load 'keys' section\n"));
-    return false;
-  }
-  if(!validate_kmxplus_header((const COMP_KMXPLUS_HEADER*)keys)) {
+bool
+COMP_KMXPLUS_KEYS::valid(KMX_DWORD _kmn_unused(length)) const {
+  KMXPLUS_PRINTF((" count: #0x%X\n", this->count));
+  if (header.size < sizeof(*this)+(sizeof(entries[0])*count)) {
```
Comment (mcdurdin) [inline core/src/kmx/kmx_plus.cpp:142]: Should this be a `!=`? Would there be a reason why `header.size` is greater than the sum of these components? Same applies in other `valid` functions :grin:

**Review by mcdurdin — APPROVED** [review]
LGTM. Let's have a discussion on forward binary compatibility at some point before we get too locked in.

File: core/src/kmx/kmx_plus.cpp (hunk 1 above)
Comment (srl295) [reply core/src/kmx/kmx_plus.cpp:142]: ooo yes.

Comment (srl295) [reply core/src/kmx/kmx_plus.cpp:142]: forward compat would be the only reason for ≠

Comment (srl295) [reply core/src/kmx/kmx_plus.cpp:142]: actually my hand assembled tiny would also have a ≠ because there's extra padding at the end of the strs.

**Pull request merged**